server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -272,16 +271,44 @@ public void onResponse(String newGeneration) {
 
                         @Override
                         public void onFailure(Exception e) {
-                            final String failure = ExceptionsHelper.stackTrace(e);
+                            final String failure;
+                            if (e instanceof AbortedSnapshotException) {
```

Comment:
I think I would prefer just to match `IndexShardSnapshotException` here and the "aborted" message (instead of introducing new exception type) to determine what to log and what failure should be.